github/git_refs.go

```diff
@@ -58,13 +60,61 @@ func (s *GitService) GetRef(ctx context.Context, owner string, repo string, ref
 
 	r := new(Reference)
 	resp, err := s.client.Do(ctx, req, r)
-	if err != nil {
+	if _, ok := err.(*json.UnmarshalTypeError); ok {
+		// Multiple refs, means there wasn't an exact match.
+		return nil, resp, nil
+	} else if err != nil {
 		return nil, resp, err
 	}
 
 	return r, resp, nil
 }
 
+// GetRefs fetches a slice of Reference objects for a given Git ref.
+// If there is an exact match, only that ref is returned.
+// If there is no exact match, GitHub returns all refs that start with ref.
+// If returned error is nil, there will be at least 1 ref returned.
+// For example:
+//
+// 	"heads/featureA" -> ["refs/heads/featureA"]                         // Exact match, single ref is returned.
+// 	"heads/feature"  -> ["refs/heads/featureA", "refs/heads/featureB"]  // All refs that start with ref.
+// 	"heads/feature"  -> []                                              // Returns an error.
```

Comment:
Maybe rename the 3rd entry to "heads/notexist" or something. Otherwise, it's confusing how "heads/feature" returns 2 branches on second line, but 0 on third.